enterprise/reporting/reporting.go

```diff
@@ -449,27 +455,32 @@ func (r *DefaultReporter) mainLoop(ctx context.Context, c types.SyncerConfig) {
 				_, err = dbHandle.Exec(`DELETE FROM `+ReportsTable+` WHERE reported_at = $1`, reportedAt)
 				if err != nil {
 					r.log.Errorf(`[ Reporting ]: Error deleting local reports from %s: %v`, ReportsTable, err)
+				} else {
+					deletedRows += len(reports)
 				}
 
 				vacuumStart := time.Now()
-				var sizeEstimate int64
-				if err := dbHandle.QueryRowContext(
-					ctx,
-					fmt.Sprintf(`SELECT pg_table_size(oid) from pg_class where relname='%s';`, ReportsTable),
-				).Scan(&sizeEstimate); err != nil {
-					r.log.Errorn(
-						`[ Reporting ]: Error getting table size estimate`,
-						logger.NewErrorField(err),
-					)
-				}
-				if sizeEstimate > config.GetInt64("Reporting.vacuumThresholdBytes", 5*bytesize.GB) {
```

Comment:
Why drop the support based on size? It sounds very useful.

Shall we make it configurable instead of dropping it?
Like, if either the size threshold or the number of records was considerable then vacuum. Wdyt?

Reply:
only removed the size constraint earlier because once the table gets to 5GB(or whatever configured value), it'll `vacuum analyze` every iteration of this loop.

Reply:
Right, we would need a `minimumInterval` to protect ourselves from this. Not hard to implement but also optional. Let's see what others think about it. If nobody gives an opinion I'm OK to drop it :+1: